tests/map_test.py

```diff
@@ -379,6 +379,23 @@ def test_evolver_remove_element():
     assert 'a' not in e
 
 
+def test_evolver_update():
```

Comment:
And this is testing that the new update method works with all the argument combinations that are supported by the builtin dict?